### PR TITLE
Document exported funcs,types,interfaces and enable check.

### PR DESCRIPTION
**Pull request by datosh**
### Proposed change(s)

- Enabled [EXC0012 & EXC0014](https://golangci-lint.run/usage/false-positives/#exc0012)
- Cleaned up all findings.

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Link to Milestone

### CODE REVIEW

**Comment by daniel-weisse** [comment]
Hey, just a heads up that some of the functions and variables that received comments have been removed from the code base since the branch was created, or will be removed in the GCP metadata refactor https://github.com/edgelesssys/constellation/pull/387 or the follow up PR for Azure

**Review by malt3 — COMMENTED** [review]
Did a preliminary review. Great work on finding proper descriptions for everything! One removed file sneaked back in.

File: internal/cloud/cloudtypes/firewall.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		return "", &PortOutOfRangeError{FromPort: fromPort, ToPort: toPort}
+	}
+	if toPort == MinPort || fromPort == toPort {
+		return fmt.Sprintf("%d", fromPort), nil
+	}
+	if toPort > MinPort {
+		return fmt.Sprintf("%d-%d", fromPort, toPort), nil
+	}
+	return "", &PortOutOfRangeError{FromPort: fromPort, ToPort: toPort}
+}
```
[the remark this answers is not in the document]
Comment (datosh) [reply internal/cloud/cloudtypes/firewall.go:151]: Thanks for letting me know 🙂 Probably when I merged `main`.

Same seems to be the case for:
* bootstrapper/internal/kubernetes/k8sapi/resources/cloudnodemanager.go
* bootstrapper/internal/kubernetes/k8sapi/resources/cluster_autoscaler.go

I have removed all three.

**Comment by datosh** [comment]
> Hey, just a heads up that some of the functions and variables that received comments have been removed from the code base since the branch was created, or will be removed in the GCP metadata refactor #387 or the follow up PR for Azure

Thanks for the heads up! The current state seems to merge fine into main. @malt3 founds a few merge problems that were resolved. 

Any concerns about merging this before #387 ?

**Comment by daniel-weisse** [comment]
Can be merged before #387 , some functions that are receiving comments will be removed by the next few PRs, but that shouldn't be a problem

File: hack/qemu-metadata-api/server/server.go [hunk 2]
```diff
@@ -21,12 +21,14 @@ import (
 	"go.uber.org/zap"
 )
 
+// Server that provides QEMU metadat.
```
Comment (derpsteb) [inline hack/qemu-metadata-api/server/server.go:24]: nit:
```suggestion
// Server that provides QEMU metadata.
```

File: bootstrapper/internal/kubernetes/k8sapi/resources/verification.go [hunk 3]
```diff
@@ -21,12 +21,14 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+// VerificationDaemonset groups all deployments for verification service.
```
Comment (derpsteb) [inline bootstrapper/internal/kubernetes/k8sapi/resources/verification.go:24]: nit: 
```suggestion
// VerificationDaemonset groups all k8s resources for verification service.
```

File: internal/attestation/azure/snp/validator.go [hunk 4]
```diff
@@ -311,6 +311,9 @@ func (a *azureInstanceInfo) validateAk(runtimeDataRaw []byte, reportData []byte,
 	return nil
 }
 
+// HCLAkValidator uses the Host Compatibility Layer, which is written by Azure,
+// and sits between Hypervisor and CVM OS as part of CVM, to validate an
+// attestation report.
```
Comment (derpsteb) [inline internal/attestation/azure/snp/validator.go:316]:
```suggestion
// HCLAkValidator validates an attestation key issued by the Host Compatibility Layer (HCL).
// The HCL is written by Azure, and sits between the Hypervisor and CVM OS.
// The HCL runs in the protected context of the CVM.
```
Shorter sentences maybe make it easier to read? :)

**Pull request merged**